packages/headless/src/controllers/knowledge/generatedAnswer/headless-knowledge-generated-answer.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  answerApi,
+  fetchAnswer,
+  selectAnswer,
+} from '../../../api/knowledge/stream-answer-api';
+import {CoreEngine} from '../../../app/engine';
+import {
+  resetAnswer,
+  updateAnswerConfigurationId,
+} from '../../../features/generated-answer/generated-answer-actions';
+import {
+  ConfigurationSection,
+  DebugSection,
+  GeneratedAnswerSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildCoreGeneratedAnswer,
+  GeneratedAnswer,
+  GeneratedAnswerAnalyticsClient,
+  GeneratedAnswerProps,
+  GeneratedResponseFormat,
+} from '../../core/generated-answer/headless-core-generated-answer';
+
+export interface KnowledgeGeneratedAnswer extends GeneratedAnswer {
+  /**
+   * Resets the last answer.
+   */
+  reset(): void;
+}
+
+interface KnowledgeGeneratedAnswerProps extends GeneratedAnswerProps {}
+
+export interface SearchAPIGeneratedAnswerAnalyticsClient
+  extends GeneratedAnswerAnalyticsClient {}
+
+export type EngineNeededForKnowledgeGeneratedAnswer = CoreEngine<
+  GeneratedAnswerSection &
+    ConfigurationSection &
+    SearchSection &
+    DebugSection & {answer: ReturnType<typeof answerApi.reducer>}
+>;
+
+/**
+ * Creates a `GeneratedAnswer` controller instance using the Answer API stream pattern.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `GeneratedAnswer` properties.
+ * @returns A `GeneratedAnswer` controller instance.
+ */
+export function buildKnowledgeGeneratedAnswer(
+  engine: EngineNeededForKnowledgeGeneratedAnswer,
+  analyticsClient: SearchAPIGeneratedAnswerAnalyticsClient,
+  props: KnowledgeGeneratedAnswerProps = {}
+): KnowledgeGeneratedAnswer {
+  if (!loadAnswerApiReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const {rephrase: coreRephrase, ...controller} = buildCoreGeneratedAnswer(
+    engine,
+    analyticsClient,
+    props
+  );
+  const getState = () => engine.state;
+
+  engine.dispatch(updateAnswerConfigurationId(props.answerConfigurationId!));
+
+  return {
+    ...controller,
+    get state() {
+      const answerApiState = selectAnswer(engine.state).data;
+      return {
+        ...getState().generatedAnswer,
+        answer: answerApiState?.answer,
+        citations: answerApiState?.citations ?? [],
+        error: {
+          message: answerApiState?.error?.message,
+          statusCode: answerApiState?.error?.code,
+        },
+        isLoading: answerApiState?.isLoading ?? false,
+        isStreaming: answerApiState?.isStreaming ?? false,
+        answerContentFormat: answerApiState?.contentFormat ?? 'text/plain',
+        isAnswerGenerated: answerApiState?.generated ?? false,
+      };
+    },
```

Comment:
I know that we intend to move from exposing the state to expose selectors. Should we do something different here ?

Reply:
Exposing state of a controller is still perfectly fine; it's the engine state what will eventually become private (it already is in the commerce engine).

Eventually we'll do some refactoring, but for now this looks good to me!